scripts/linters/custom_eslint_checks/rules/test-message-style.js

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint check to ensure that correct style is followed for the
+ * test messages in calling of 'it'.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'There should not be any unused directive',
+      category: 'Stylistic Issues',
+      recommended: false
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      useShould: 'Test message should start with \'should\'',
+      singleSpace: 'Do not use multiple consecutive spaces in the test message',
+      noSpaceAtEnd: 'Do not use space at the end of test message'
+    }
+  },
+
+  create: function(context) {
+    var nodePos = {};
+    var testMessage = '';
+
+    var checkMessageStartsWithShould = function(testMessageNode) {
+      if (!testMessage.startsWith('should ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'useShould'
+        });
+      }
+    };
+
+    var checkSpacesInMessage = function(testMessageNode) {
+      if (testMessage.includes('  ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'singleSpace'
+        });
+      }
+    };
+
+    var checkNoSpaceAtEndOfMessage = function(testMessageNode) {
+      if (testMessage.endsWith(' ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'noSpaceAtEnd'
+        });
+      }
+    };
+
+    var checkMessage = function(node) {
+      checkMessageStartsWithShould(node);
+      checkNoSpaceAtEndOfMessage(node);
+      checkSpacesInMessage(node);
+    };
+
+    var extractMessage = function(node) {
+      if (node.type === 'Literal') {
+        testMessage += node.value;
+        if (node.loc.end.line === nodePos.end.line &&
+            node.loc.end.column === nodePos.end.column) {
+          checkMessage(node);
+        }
+      } else if (node.type === 'BinaryExpression') {
+        extractMessage(node.left);
+        extractMessage(node.right);
+      }
+    };
```

Comment:
Should we do it this way?

```
    var extractMessage = function(node) {
      if (node.type === 'Literal') {
        return node.value;
      } else if (node.type === 'BinaryExpression') {
        return extractMessage(node.left) + extractMessage(node.right);
      }
    };


    return {
      CallExpression(node) {
        if (node.callee.name === 'it') {
          const testMessageNode = node.arguments[0];
          testMessage = extractMessage(testMessageNode);
          checkMessage(testMessage);
        }
      }
```

Reason for suggesting this structure:
1. The current structure has a function named `extractMessage` which is not extracting the message correctly + it's also running the checks which are unexpected.
2. `extractMessage` is populating a non-local variable that is being used by other checker functions (i.e, all checker functions are using non-local variable).
3. The logic inside `extractMessage` function around on line 82 doesn't seem simple and clear.

Reply:
Done.